Lib/test/test_funcattrs.py

```diff
@@ -98,7 +98,19 @@ def test___globals__(self):
                              (AttributeError, TypeError))
 
     def test___builtins__(self):
-        self.assertIs(self.b.__builtins__, __builtins__)
+        if __name__ == "__main__":
+            # When this test is run by executing the current module,
+            # `__builtins__` is the built-in module `builtins`.
+
+            # When this test is run from another module, `__builtins__` is
+            # `builtins.__dict__`
+
+            import builtins
+            builtins_dict = builtins.__dict__
+        else:
+            builtins_dict = __builtins__
```

Comment:
I am not sure that the comment is useful. You can simply the code to:

```suggestion
        if __name__ == "__main__":
            builtins_dict = __builtins__.__dict__
        else:
            builtins_dict = __builtins__
```

Reply:
Looks good. Thank you!